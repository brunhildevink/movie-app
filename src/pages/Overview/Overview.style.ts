import styled, { keyframes } from 'styled-components'
import { colors } from '../../styles'

const App = styled.div`
  text-align: center;
`

const Code = styled.code`
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
`

const AppHeader = styled.header`
  background-color: ${colors.black};
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
`

const Spin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const Logo = styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${Spin} infinite 20s linear;
  }
`

const AppLink = styled.a`
  color: ${colors.blue};
`

export { App, Code, AppHeader, Logo, AppLink }
