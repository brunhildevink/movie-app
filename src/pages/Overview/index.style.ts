import styled from 'styled-components'
import { breakpoints } from '../../styles'

const Container = styled.div`
  @media ${breakpoints.lg} {
    display: grid;
    grid-template-columns: 60% 40%;
    height: 100%;
    overflow: hidden;
  }
`

const Wrapper = styled.div`
  @media ${breakpoints.lg} {
    height: 100vh;
  }
`
export { Container, Wrapper }
