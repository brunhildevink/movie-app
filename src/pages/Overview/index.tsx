import React, { useEffect, useState } from 'react'
import useFetch from '../../hooks/useFetch'
import { FetchParams, MotionPicture } from '../../lib/types'
import { App, AppHeader } from './Overview.style'
import { Heading } from '../../components'
import formatUrl from '../../utils/helpers/formatUrl'

const Header: React.FC = () => {
  const [fetchParams, setFetchParams] = useState<FetchParams>({
    t: 'mandalorian',
    plot: 'full',
    season: 1,
  })

  const { data, error, loading, refetch, updateUrl } = useFetch<MotionPicture>('/', fetchParams)

  const totalSeasons = data ? parseInt(data.totalSeasons, 10) : 0

  useEffect(() => {
    updateUrl(formatUrl('/', fetchParams))
    refetch()
  }, [fetchParams])

  useEffect(() => {
    console.log(data)
  }, [data])

  const handleChange = (currentSeason: number) => {
    const newParams: FetchParams = { ...fetchParams }
    newParams.season = currentSeason
    setFetchParams(newParams)
  }

  const renderSelectSeasons = Array.from(Array(totalSeasons).keys()).map((num) => (
    <option key={num} value={num + 1}>
      Season {num + 1}
    </option>
  ))

  return (
    <App>
      <AppHeader>
        {loading && <Heading.HeadingFour>Loading...</Heading.HeadingFour>}

        {error && <Heading.HeadingFour>Oh no, something happened...</Heading.HeadingFour>}

        {data && (
          <select
            disabled={!(totalSeasons > 0)}
            key={totalSeasons}
            onChange={(event) => handleChange(parseInt(event.target.value, 10))}
          >
            {renderSelectSeasons}
          </select>
        )}
      </AppHeader>
    </App>
  )
}

export default Header
