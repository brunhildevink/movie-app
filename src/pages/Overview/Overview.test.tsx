import React from 'react'
import renderer from 'react-test-renderer'
import Overview from '.'

test('Header renders correctly', () => {
  const tree = renderer.create(<Overview />).toJSON()
  expect(tree).toMatchSnapshot()
})
