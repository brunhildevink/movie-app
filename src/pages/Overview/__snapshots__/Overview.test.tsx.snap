// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="sc-bczRLJ kIOTIl"
>
  <header
    className="sc-dkzDqf heTzSB"
  />
</div>
`;
